From b3810efb8bd88ba65b7dc9d8a080eefe69029994 Mon Sep 17 00:00:00 2001
From: Ihor Usyk <ihor_usyk@epam.com>
Date: Wed, 15 Feb 2023 21:29:08 +0100
Subject: [PATCH 17/18] aquire_scmi_channel: code optimization

The code used the bool variable 'found' was used
to identify that channel had been found. The additional condition was used
to return the required result, like
if ( found )
    return curr;
return NULL;

It is proposed to user variable 'ret' and remove the condition;
like:
int *ret = NULL;
/* here the loop and if value is found - assign value to ret*/
return ret;

Signed-off-by: Ihor Usyk <ihor_usyk@epam.com>
---
 xen/arch/arm/sci/scmi_smc.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/xen/arch/arm/sci/scmi_smc.c b/xen/arch/arm/sci/scmi_smc.c
index 103529dfab..b389a6931d 100644
--- a/xen/arch/arm/sci/scmi_smc.c
+++ b/xen/arch/arm/sci/scmi_smc.c
@@ -383,7 +383,7 @@ static struct scmi_channel *get_channel_by_id(uint8_t chan_id)
 static struct scmi_channel *aquire_scmi_channel(domid_t domain_id)
 {
     struct scmi_channel *curr;
-    bool found = false;
+    struct scmi_channel *ret = NULL;
 
     ASSERT(domain_id != DOMID_INVALID && domain_id >= 0);
 
@@ -393,16 +393,14 @@ static struct scmi_channel *aquire_scmi_channel(domid_t domain_id)
         if ( curr->domain_id == DOMID_INVALID )
         {
             curr->domain_id = domain_id;
-            found = true;
+            ret = curr;
             break;
         }
     }
 
     spin_unlock(&scmi_data.channel_list_lock);
-    if ( found )
-        return curr;
 
-    return NULL;
+    return ret;
 }
 
 static void relinquish_scmi_channel(struct scmi_channel *channel)
-- 
2.25.1

